fastly/fastly_alerts.go

```diff
@@ -0,0 +1,344 @@
+package fastly
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+// AlertDefinition holds the configuration for one alert.
+type AlertDefinition struct {
+	ID                 string              `json:"id"`
```

Comment:
I'm a little undecided on the use of `json` struct tags vs `mapstructure` (https://github.com/mitchellh/mapstructure). 

On the one hand I feel like we should use mapstructure for consistency, because it is used currently throughout the entire codebase. 

But I also am not a fan of the 'magic' that is mapstructure 🙂 it's also an extra level of complexity in the code design that would be nice to avoid.

It's also never been overly clear to me why it was necessary to use it in the first place. I presume it's probably related to our API returning data types that aren't quite right for the field. For example, the API returns, what otherwise should be an integer, as a string. Using `mapstructure` means we're able to more easily marshal the data into the format we want it to be _exposed_ as (as part of the go-fastly interface) to the customer, rather than having to implement our own custom marshaler for each API endpoint.

But there has been a discussion internally at Fastly regarding the removal/replacement of mapstructure because it would help with the support of go-fastly on Fastly's Compute platform. So using native `json` unmarshaling here would help start us off in that direction.

So at the moment I'm in favour of using `json` but I might need to sit and dwell on it for a bit longer.

Reply:
OK having given this some thought I think it's OK to go ahead with `json` tags as I'm inclined to want to replace mapstructure in the future. We have an internal ticket tracking this 'replace mapstructure' request and I've commented there also.

Reply:
Thanks for the detailed explanation.